src/serializer/ResidualHeapSerializer.js

```diff
@@ -1789,14 +1793,16 @@ export class ResidualHeapSerializer {
     return false;
   }
 
-  _serializeAdditionalFunctionGeneratorAndEffects(generator: Generator, postGeneratorCallback: () => void) {
-    let context = this._getContext();
+  _serializeAdditionalFunctionGeneratorAndEffects(generator: Generator, additionalEffects: AdditionalFunctionEffects) {
     return this._withGeneratorScope("AdditionalFunction", generator, newBody => {
       let oldSerialiedValueWithIdentifiers = this._serializedValueWithIdentifiers;
       this._serializedValueWithIdentifiers = new Set(Array.from(this._serializedValueWithIdentifiers));
       try {
-        generator.serialize(context);
-        postGeneratorCallback();
+        let effectsGenerator = additionalEffects.generator;
+        effectsGenerator.serialize(this._getContext());
+
+        const lazyHoistedReactNodes = this.residualReactElementSerializer.serializeLazyHoistedNodes();
+        Array.prototype.push.apply(this.mainBody.entries, lazyHoistedReactNodes);
```

Comment:
Why apply push rather than calling it directly?

Reply:
Because lazyHoistedReactNodes is an array I guess.

Reply:
You can do `this.mainBody.entries.push(...lazyHoistedReactNodes)` instead.